File: Examples/StateDiagramDiff/Swift/After/Download.swift
/// The phases a download moves through.
public enum DownloadState {
    case idle
    case requested
    case downloading
    case verifying
    case finished
    case failed
}

/// A download whose `state` advances through a pipeline. `run()` walks the happy path as a
/// sequence of assignments — which the value-flow analysis renders as a transition chain —
/// while `fail()` is a branch reachable from the start.
public final class Download {
    public private(set) var state: DownloadState = .idle

    public func run() {
        state = .requested
        state = .downloading
        state = .verifying
        state = .finished
    }

    public func fail() {
        state = .failed
    }
}

File: Examples/StateDiagramDiff/TypeScript/After/Download.ts
/** The phases a download moves through. */
export enum DownloadState {
    Idle,
    Requested,
    Downloading,
    Verifying,
    Finished,
    Failed,
}

/** A download whose `state` advances through a pipeline: `run()` walks the happy path as a
 *  sequence of assignments (a transition chain), while `fail()` branches from the start. */
export class Download {
    private state: DownloadState = DownloadState.Idle;

    run(): void {
        this.state = DownloadState.Requested;
        this.state = DownloadState.Downloading;
        this.state = DownloadState.Verifying;
        this.state = DownloadState.Finished;
    }

    fail(): void {
        this.state = DownloadState.Failed;
    }
}
